src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -199,14 +199,24 @@ Task IHttpResponseBodyFeature.SendFileAsync(string path, long offset, long? coun
 
         // TODO: In the future this could complete the body all the way down to the server. For now it just ensures
         // any unflushed data gets flushed.
-        protected Task CompleteResponseBodyAsync()
+        protected async Task CompleteResponseBodyAsync()
```

Comment:
Is this called every request? Would it be worth it to fast-path this when all the returned (Value)Tasks are completed?